public/app/features/dashboard-scene/panel-edit/VizPanelManager.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+
+import { FieldConfigSource, PanelModel } from '@grafana/data';
+import {
+  SceneObjectState,
+  VizPanel,
+  SceneObjectBase,
+  SceneComponentProps,
+  sceneUtils,
+  DeepPartial,
+} from '@grafana/scenes';
+
+interface VizPanelManagerState extends SceneObjectState {
+  panel: VizPanel;
+}
+
+export class VizPanelManager extends SceneObjectBase<VizPanelManagerState> {
```

Comment:
Think this is an interesting idea. Do you think this would serve as a universal API for other VizPanel edit behaviors?

Think going forward we could scale this to have QueriesManager / TransformationsManager as well so that those aspects of panel edit are abstracted away and not all implemented in this  single class.

Would love to see how we implement `Apply` option with this abstraction - will most probably be just replacing the original panel in the scene with `VizPanelManager.state.panel`.